python/eight_mile/tf/layers.py

```diff
@@ -1544,3 +1575,404 @@ def with_device(*args, **kwargs):
             return func(*args, **kwargs)
         return func(*args, **kwargs)
     return with_device
+
+
+class BaseAttention(tf.keras.layers.Layer):
```

Comment:
Have there been any papers on if using the kqv form of attention works better for this sort of vector to sequence attention used in seq2seq?